src/components/Guilds/ProposalPage/ProposalInfoCard/index.tsx

```diff
@@ -34,7 +38,40 @@ const UserInfoDetail = styled(Box)`
   font-weight: 600;
 `;
 
-const ProposalInfoCard = () => {
+interface ProposalInfoCardProps {
+  proposalId: string;
+  guildId: string;
+}
+
+const ProposalInfoCard: React.FC<ProposalInfoCardProps> = ({
+  guildId,
+  proposalId,
+}) => {
+  const { proposal, error } = useProposal(guildId, proposalId);
+
+  const startTime = useMemo(() => {

```

Comment:
@mprasanjith This stuff is very similar to ProposalStatus calculation. 
Seems to me we can add this memo as part of a middleware in useEtherSWR, so that every result data is postprocessed and the attribute calculated and added, OR, we can put this as a helper in useProposal hook, or as an attribute for every proposal as part of its prototype.